### PR TITLE
refactor:  replace laminas-mvc-console with laminas-cli

### PR DESCRIPTION
**Pull request by fibble**
## Description

Migrate laminas-mvc-console commands to laminas-cli

Related issue: [VOL-3615](https://dvsa.atlassian.net/browse/VOL-3615)

Requires simultaneous merge of olcs-logging && olcs-backend or will break builds.

## Before submitting (or marking as "ready for review")

- [x] Does the pull request title follow the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification?
- [x] Have you performed a self-review of the code
- [x] Have you have added tests that prove the fix or feature is effective and working
- [ ] Did you make sure to update any documentation relating to this change?

### CODE REVIEW

File: src/Listener/LogRequest.php [hunk 1]
```diff
@@ -40,8 +41,8 @@ public function onRoute(MvcEvent $e)
     {
         if ($this->isConsole($e)) {
             $data = [
-                'path' => $e->getRequest()->getScriptName(),
-                'params' => $e->getRequest()->getParams()
+                'path' => $_SERVER['argv'][0],
+                'params' => array_slice($_SERVER['argv'], 1)
```
Comment (JoshuaLicense) [inline src/Listener/LogRequest.php:45]: Can we put these to come from methods via. the interface instead, please?

File: src/CliLoggableInterface.php [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?php
+
+namespace Olcs\Logging;
+
+interface CliLoggableInterface {
+}
```
Comment (JoshuaLicense) [inline src/CliLoggableInterface.php:6]: Would be good to define some methods for static analysis, `getScriptName` & `getParams` at least.

Can use: https://github.com/laminas/laminas-console/blob/master/src/Request.php as a reference of how Laminas did this originally - we could do with something similar.

File: src/Request/CliRequest.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    }
+
+    public function setContent($content)
+    {
+    }
+
+    public function getContent()
+    {
+    }
+}
```
Comment (JoshuaLicense) [inline src/Request/CliRequest.php:25]: This belongs in the `olcs-backend` repository. How come we don't extend: `\Laminas\Stdlib\Message` instead too?

**Review by JoshuaLicense — CHANGES_REQUESTED** [review]

File: src/Listener/LogRequest.php (hunk 1 above)
Comment (fibble) [reply src/Listener/LogRequest.php]: Done

File: src/CliLoggableInterface.php (hunk 2 above)
Comment (fibble) [reply src/CliLoggableInterface.php]: Added thanks

File: src/Request/CliRequest.php (hunk 3 above)
Comment (fibble) [reply src/Request/CliRequest.php]: Removed.

File: test/Listener/LogRequestTest.php [hunk 4]
```diff
@@ -2,16 +2,30 @@
...
+    {
+        parent::setUp();
+        $this->originalArgv = $_SERVER['argv'];
+    }
+
+    protected function tearDown(): void
+    {
+        $_SERVER['argv'] = $this->originalArgv;
+        parent::tearDown();
+    }
```
Comment (JoshuaLicense) [inline test/Listener/LogRequestTest.php:27]: Do we still need these?

File: test/Listener/LogRequestTest.php [hunk 5]
```diff
@@ -270,12 +284,11 @@ public function testHttpOnDispatch()
...
-        $scriptName = 'file.php';
-        $params = ['route-name', '--help'];
+        $_SERVER['argv'] = ['file.php', 'route-name', '--help'];
 
-        $mockRequest = m::mock('Laminas\Console\Request');
-        $mockRequest->shouldReceive('getScriptName')->andReturn($scriptName);
-        $mockRequest->shouldReceive('getParams')->andReturn($params);
+        $mockRequest = m::mock(CliLoggableInterface::class);
+        $mockRequest->shouldReceive('getScriptPath')->andReturn($_SERVER['argv'][0]);
+        $mockRequest->shouldReceive('getScriptParams')->andReturn(array_slice($_SERVER['argv'], 1));
```
Comment (JoshuaLicense) [inline test/Listener/LogRequestTest.php:291]: Do we need to use globals?

File: test/Listener/LogRequestTest.php [hunk 6]
```diff
@@ -285,20 +298,27 @@ public function testConsoleOnDispatch()
             'Request received',
             [
                 'data' => [
-                    'path' => $scriptName,
-                    'params' => $params,
+                    'path' => $_SERVER['argv'][0],
+                    'params' => array_slice($_SERVER['argv'], 1),
```
Comment (JoshuaLicense) [inline test/Listener/LogRequestTest.php:302]: Same as above.

**Review by JoshuaLicense — CHANGES_REQUESTED** [review]

**Review by JoshuaLicense — APPROVED** [review]

**Pull request merged**